include/nektar_interface/geometry_transport_3d.hpp

```diff
@@ -0,0 +1,586 @@
+      ASSERTL0(face_ptr != nullptr, "No face in element map.");
+      ASSERTL0(face_ptr->GetGlobalID() == face_id, "Element id missmatch.");
+      faces[facex] = face_ptr;
+    }
+    auto new_geom = std::make_shared<PrismGeom>(geom_id, faces.data());
+    new_geom->GetGeomFactors();
+    new_geom->Setup();
+    *base_ptr = ptr;
+    return std::make_shared<RemoteGeom3D>(rank, geom_id, new_geom);
+  }
+  case ShapeType::ePyramid: {
+    std::vector<Geometry2DSharedPtr> faces(5);
+    for (int facex = 0; facex < 5; facex++) {
+      const int face_id = *ptr++;
+      auto face_ptr = element_map[face_id];
+      ASSERTL0(face_ptr != nullptr, "No face in element map.");
+      ASSERTL0(face_ptr->GetGlobalID() == face_id, "Element id missmatch.");
+      faces[facex] = face_ptr;
+    }
+    auto new_geom = std::make_shared<PyrGeom>(geom_id, faces.data());
+    new_geom->GetGeomFactors();
+    new_geom->Setup();
+    *base_ptr = ptr;
+    return std::make_shared<RemoteGeom3D>(rank, geom_id, new_geom);
+  }
+  case ShapeType::eTetrahedron: {
+    std::vector<TriGeomSharedPtr> faces(4);
+    for (int facex = 0; facex < 4; facex++) {
+      const int face_id = *ptr++;
+      auto face_ptr = element_map[face_id];
+      ASSERTL0(face_ptr != nullptr, "No face in element map.");
+      ASSERTL0(face_ptr->GetGlobalID() == face_id, "Element id missmatch.");
+      ASSERTL0(face_ptr->GetShapeType() == ShapeType::eTriangle,
+               "Element type missmatch.");
+      faces[facex] = std::dynamic_pointer_cast<TriGeom>(face_ptr);
+    }
+    auto new_geom = std::make_shared<TetGeom>(geom_id, faces.data());
+    new_geom->GetGeomFactors();
+    new_geom->Setup();
+    *base_ptr = ptr;
+    return std::make_shared<RemoteGeom3D>(rank, geom_id, new_geom);
+  }
+  default: {
+    return nullptr;
+  }
+  }
+}
+
+/**
+ * Reconstruct multiple 3D geometry objects from a vector of ints.
+ *
+ * @param[in] rank_element_map_2d Map from MPI rank to element map (geom id to
+ * shared ptr) required to rebuild geoms.
+ * @param[in] packed_geoms Int vector of 3D geometry objects described by 2D
+ * geometry ids.
+ * @param[in,out] output_container Output vector in which to place constructed
+ * 3D geometry objects.
+ */
+inline void reconstruct_geoms_3d(
+    std::map<int,
+             std::map<int, std::shared_ptr<Nektar::SpatialDomains::Geometry2D>>>
+        &rank_element_map_2d,
+    std::vector<int> &packed_geoms,
+    std::vector<std::shared_ptr<RemoteGeom3D>> &output_container) {
+
+  // rebuild the 3D geometry objects
+  int *base_ptr = packed_geoms.data();
+  int *end_ptr = base_ptr + packed_geoms.size();
+  while (base_ptr < end_ptr) {
+    auto new_geom_3d = reconstruct_geom_3d(&base_ptr, rank_element_map_2d);
+    ASSERTL0(new_geom_3d != nullptr,
+             "Could not recreate a 3D geometry object.");
+    output_container.push_back(new_geom_3d);
+  }
+}
+
+/**
+ * Desconstruct a 3D geometry object into a series of integers that describe
```

Comment:
```suggestion
 * Deconstruct a 3D geometry object into a series of integers that describe
```
At least one more example below; suggest find-replace